network/wsNetwork.go

```diff
@@ -1146,10 +1170,22 @@ func (wn *WebsocketNetwork) ServeHTTP(response http.ResponseWriter, request *htt
 		prioChallenge:     challenge,
 		createTime:        trackedRequest.created,
 		version:           matchingVersion,
+		identity:          peerPublicKey,
+		// record the challenge we gave to the initiating peer so we can verify
+		identityChallenge: idChalResp.ResponseChallenge,
+		identityVerified:  false,
 	}
 	peer.TelemetryGUID = trackedRequest.otherTelemetryGUID
 	peer.init(wn.config, wn.outgoingMessagesBufferSize)
 	wn.addPeer(peer)
+	wn.peersByID[peerPublicKey] = peer
+	// start a thread to disconnect the peer if it has not verified its identity in time
+	go func() {
+		time.Sleep(5 * time.Second)
+		if !peer.identityVerified {
+			wn.Disconnect(peer)
+		}
+	}()
```

Comment:
factor this into its own function in wsIdentity.go?

Reply:
I imagine this might look better as a member function on the peer object, then the line here would be `go peer.waitForIdentityVerify()` and from there it could follow the peer self-close path

Reply:
I like it, thanks :)